cmd/nodecmd/create.go

```diff
@@ -0,0 +1,678 @@
+	if err != nil {
+		return "", err
+	}
+	certFilePath := homeDir + "/.ssh/" + certName
+	return certFilePath, nil
+}
+
+func createNodeConfig(nodeID, region, ami, keyPairName, certPath, sg, eip, clusterName string) error {
+	elasticIPToUse := eip[1 : len(eip)-2]
+	nodeIDToUse := nodeID[1 : len(nodeID)-2]
+
+	nodeConfig := models.NodeConfig{
+		NodeID:        nodeIDToUse,
+		Region:        region,
+		AMI:           ami,
+		KeyPair:       keyPairName,
+		CertPath:      certPath,
+		SecurityGroup: sg,
+		ElasticIP:     elasticIPToUse,
+	}
+	err := app.CreateNodeConfigFile(nodeIDToUse, &nodeConfig)
+	if err != nil {
+		return err
+	}
+	return updateClusterConfig(nodeIDToUse, keyPairName, certPath, clusterName)
+}
+
+func updateClusterConfig(nodeID, keyPairName, certPath, clusterName string) error {
+	configExists := app.ClusterConfigExists()
+	clusterConfig := models.ClusterConfig{}
+	var err error
+	if configExists {
+		clusterConfig, err = app.LoadClusterConfig()
+		if err != nil {
+			return err
+		}
+	}
+
+	if clusterConfig.KeyPair == nil {
+		clusterConfig.KeyPair = make(map[string]string)
+	}
+	if _, ok := clusterConfig.KeyPair[keyPairName]; !ok {
+		clusterConfig.KeyPair[keyPairName] = certPath
+	}
+	if clusterConfig.Clusters == nil {
+		clusterConfig.Clusters = make(map[string][]string)
+	}
+	if _, ok := clusterConfig.Clusters[clusterName]; !ok {
+		clusterConfig.Clusters[clusterName] = []string{}
+	}
+	clusterConfig.Clusters[clusterName] = append(clusterConfig.Clusters[clusterName], nodeID)
+	return app.UpdateClusterConfig(&clusterConfig)
+}
+
+func printNoCredentialsOutput() {
+	ux.Logger.PrintToUser("No AWS credentials file found in ~/.aws/credentials")
+	ux.Logger.PrintToUser("Create a file called 'credentials' with the contents below, and add the file to ~/.aws/ directory")
+	ux.Logger.PrintToUser("===========BEGINNING OF FILE===========")
+	ux.Logger.PrintToUser("[default]\naws_access_key_id=<AWS_ACCESS_KEY>\naws_secret_access_key=<AWS_SECRET_ACCESS_KEY>")
+	ux.Logger.PrintToUser("===========END OF FILE===========")
+	ux.Logger.PrintToUser("More info can be found at https://docs.aws.amazon.com/sdkref/latest/guide/file-format.html#file-format-creds")
+}
+
+func createNode(_ *cobra.Command, args []string) error {
+	clusterName := args[0]
+
+	err := removeExistingTerraformFiles()
+	if err != nil {
+		return err
+	}
+	usr, _ := user.Current()
+	keyPairName := usr.Username + "-avalanche-cli"
+	certName := keyPairName + "-kp.pem"
+	securityGroupName := keyPairName + "-sg"
+	region := "us-east-2"
+	ami := "ami-0430580de6244e02e"
+
+	hclFile := hclwrite.NewEmptyFile()
+	tfFile, err := os.Create("node_config.tf")
+	if err != nil {
+		return err
+	}
+	rootBody := hclFile.Body()
+	creds := credentials.NewSharedCredentials("", "default")
+	credValue, err := creds.Get()
+	if err != nil {
+		printNoCredentialsOutput()
+		return err
+	}
+	err = setCloudCredentials(rootBody, credValue.AccessKeyID, credValue.SecretAccessKey, region)
```

Comment:
probably all stuff related to creating the terraform config file should have its own function, and probably this aux functions may also indicate that they are updating the terraform config file